serde_v8/magic/detached_buffer.rs

```diff
@@ -1,41 +1,49 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
-use core::ops::Range;
-use std::ops::Deref;
-use std::ops::DerefMut;
-
 use super::transl8::FromV8;
 use super::transl8::ToV8;
-use super::v8slice::to_ranged_buffer;
+use super::v8slice::value_to_array_buffer;
 use super::v8slice::V8Slice;
 use crate::magic::transl8::impl_magic;
 
 // A buffer that detaches when deserialized from JS
 pub struct DetachedBuffer(V8Slice);
 impl_magic!(DetachedBuffer);
 
-impl AsRef<[u8]> for DetachedBuffer {
-  fn as_ref(&self) -> &[u8] {
-    self.0.as_ref()
-  }
-}
-
-impl AsMut<[u8]> for DetachedBuffer {
-  fn as_mut(&mut self) -> &mut [u8] {
-    self.0.as_mut()
+impl DetachedBuffer {
```

Comment:
It seems like the `V8Slice` constraints don't make sense for `DetachedBuffer`: since ABs uniquely own their backing memory (unlike SABs) and the only way to obtain a `DetachedBuffer` is by detaching an AB, it should be perfectly safe to get slices to the backing store memory, as long as the Rust memory model is otherwise followed.